pattoo/db/schema/user.py

```diff
@@ -75,34 +65,54 @@ class Meta:
         # Hide certain fields as a tuple
         exclude_fields = ('password', )
 
+class ProtectedUser(graphene.Union):
+    class Meta:
+        types = (User, AuthInfoField)
+
 
-class CreateUserInput(graphene.InputObjectType, UserAttributeWithPassword):
+class CreateUserInput(graphene.InputObjectType, UserAttribute):
     """Arguments to create a User."""
-    pass
+
+    password = graphene.String(description="Password.")
 
 
 class CreateUser(graphene.Mutation):
     """Create a User Mutation."""
 
-    user = graphene.Field(
-        lambda: User, description='User created by this mutation.')
+    user = graphene.Field(lambda: ProtectedUser,
+                          description='User created by this mutation.')
 
     class Arguments:
         Input = CreateUserInput(required=True)
+        token = graphene.String()
+
+    @classmethod
+    @mutation_jwt_required
+    def mutate(cls, _, info_, Input):
 
-    def mutate(self, info_, Input):
         data = _input_to_dictionary(Input)
         user = UserModel(**data)
+        token_username = get_jwt_identity()
 
-        # Create user only if they don't already exist
+        # Getting current user making request to resource
+        current_user = UserTable(token_username)
+
+        # Accessing user data
         person = UserTable(user.username.decode())
-        if person.exists is False:
-            with db.db_modify(20150, close=False) as session:
-                session.add(user)
-            return CreateUser(user=user)
 
-        # Return nothing otherwise
-        return None
+        # Checking that the user creating the new user is an admin
+        if current_user.role != 0:
+            raise GraphQLError('Only admins can create a new user!')
+
+        # Checking that a given username is not taken
+        if person.exists:
+            raise GraphQLError('Username already exists!')
```

Comment:
Also use `pattoo_shared.log2die`, that way the message appears in the logs. This won't allow proper troubleshooting.

Reply:
@Dev-Dominic